extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -428,23 +428,54 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
 	private async openConsentDialog(tenantId: string, resourceId: string): Promise<boolean> {
+		const getTenantConfigurationSet = (): Set<string> => {
+			const configuration = vscode.workspace.getConfiguration('azure.tenant.config');
+			let values: string[] = configuration.get('filter') ?? [];
+			return new Set<string>(values);
+		};
+
+		const updateTenantConfigurationSet = (set: Set<string>): void => {
+			const configuration = vscode.workspace.getConfiguration('azure.tenant.config');
+			configuration.update('filter', Array.from(set), vscode.ConfigurationTarget.Global);
+		};
+
+		// The user wants to ignore this tenant.
+		if (getTenantConfigurationSet().has(tenantId)) {
+			return false;
+		}
+
 		interface ConsentMessageItem extends vscode.MessageItem {
 			booleanResult: boolean;
+			action?: (tenantId: string) => void;
 		}
 
 		const openItem: ConsentMessageItem = {
-			title: localize('open', "Open"),
+			title: localize('azurecore.consentDialog.open', "Open"),
 			booleanResult: true
 		};
 
 		const closeItem: ConsentMessageItem = {
-			title: localize('cancel', "Cancel"),
+			title: localize('azurecore.consentDialog.cancel', "Cancel"),
 			isCloseAffordance: true,
 			booleanResult: false
 		};
 
+		const dontAskAgainItem: ConsentMessageItem = {
+			title: localize('azurecore.consentDialog.noAsk', "Don't Ask Again"),
```

Comment:
"Don't ask again" doesn't indicate to me that it's going to actually ignore this tenant. Maybe something like like `Ignore` and then add a sentence in the box about how ignore will skip this tenant until it's removed from the filter. I can't think of something that would fit comfortably in the button itself that would get the point across.

Reply:
What about when someone presses ignore they get a nice notification that says what happened?

Reply:
you can say something like: "please press 'Ignore tenant' button if you don't want to get notification about this tenant"